### ISSUE TITLE
type declaration with vararg tuple fails to parse parameterized type

### ISSUE THREAD

**Issue by vshesh** [DanielXMoore/Civet#519]
https://civet.dev/playground?code=dHlwZSBUZXN0PFM%2BID0gbnVtYmVyIHwgc3RyaW5nIHwgUzsKdHlwZSBUZXN0MiA9IFtzdHJpbmcsIC4uLlRlc3Q8Uz5d

basically this parses fine:
```
type Test = [string, string, ...string]
```

but this does not: 
```
type Test = [string, string, ...Array<string>]
```

and neither does this 
```
type Test = [string, ...string[]]
```

All of these work in typescript: 
```ts
type Test = [string, ...string[]] | [string, ...Array<string>]
var x : Test = ['', '', '']
```

is there a way to get civet to "pass through"  this type declaration to typescript?

**Comment by vshesh** [comment]
error: 
```
Failed to parse
Expected:
	OpenParen "("
	Whitespace /[\s]+/
	JSMultiLineComment "/*"
	CoffeeHereCommentStart /###(?!#)/
	JSSingleLineComment /\/\/(?!\/)[^\r\n]*/
	ImplementsShorthand "<:"
	ImplementsToken "implements"
	NonNewlineWhitespace /[ \t]+/
	EOL /\r\n|\n|\r|$/
	InlineComment "/*"
Found: "[]]"
```

**Comment by edemaine** [comment]
It looks like `TypeTuple` doesn't support spreads. I'll take a look now.

**Comment by vshesh** [comment]
Thanks for fixing the bug. 
How can I get the fix? I need this kind of type signature for the project I'm doing...

**Comment by STRd6** [comment]
Published in v0.6.2 just now, you should be able to update!

Thanks for opening the issue!
